### PR TITLE
Ub fixes

### PR DESCRIPTION
**Pull request by syyyr**

### CODE REVIEW

**Comment by syyyr** [comment]
Needs tests... I'll add them later

**Comment by syyyr** [comment]
Ok, now have tests. @fvacek If you think you can simplify the fromChainpack test, feel free to do so, I have just copied the input from the fuzzer.

File: libshvchainpack/fuzz/fromChainpack.cpp [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
     std::string err;
-    auto rpc_value = shv::chainpack::RpcValue::fromChainPack({reinterpret_cast<const char*>(data), size}, &err);
+    try {
+        auto rpc_value = shv::chainpack::RpcValue::fromChainPack({reinterpret_cast<const char*>(data), size}, &err);
```
Comment (fvacek) [inline libshvchainpack/fuzz/fromChainpack.cpp:6]: `shv::chainpack::RpcValue::fromChainPack()` should not throw if `err` is provided

File: libshvchainpack/src/chainpack/rpcvalue.cpp [hunk 2]
```diff
@@ -165,13 +165,25 @@ class ValueData : public RpcValue::AbstractValueData
...
 class ChainPackDouble final : public ValueData<RpcValue::Type::Double, double>
 {
 	ChainPackDouble* create() override { return new ChainPackDouble(0); }
 	std::string toStdString() const override { return std::to_string(m_value); }
 	double toDouble() const override { return m_value; }
 	bool toBool() const override { return !(m_value == 0.); }
-	RpcValue::Int toInt() const override { return static_cast<RpcValue::Int>(m_value); }
+	RpcValue::Int toInt() const override {
+		if (!is_representable_as_int(m_value)) {
+			throw std::runtime_error("This double is not representable as int");
```
Comment (fvacek) [inline libshvchainpack/src/chainpack/rpcvalue.cpp:183]: I'm not sure, if I adding `throw` to `to*()` is good idea, what about return `nan`

**Pull request merged**